packages/js/src/dashboard/scores/components/scores.js

```diff
@@ -15,105 +15,99 @@ import { TermFilter } from "./term-filter";
+	} catch ( e ) {
+		return sprintf( message, "", "" );
 	}
-
-	return url;
 };
 
-// Added dummy space as content to prevent children prop warnings in the console.
-const supportLink = <Link variant="error" href="admin.php?page=wpseo_page_support"> </Link>;
-
-const TimeoutErrorMessage = createInterpolateElement(
-	sprintf(
-		/* translators: %1$s and %2$s expand to an opening/closing tag for a link to the support page. */
-		__( "A timeout occurred, possibly due to a large number of posts or terms. In case you need further help, please take a look at our %1$sSupport page%2$s.", "wordpress-seo" ),
-		"<supportLink>",
-		"</supportLink>"
-	),
-	{
-		supportLink,
-	}
-);
-const OtherErrorMessage = createInterpolateElement(
-	sprintf(
-		/* translators: %1$s and %2$s expand to an opening/closing tag for a link to the support page. */
-		__( "Something went wrong. In case you need further help, please take a look at our %1$sSupport page%2$s.", "wordpress-seo" ),
-		"<supportLink>",
-		"</supportLink>"
-	),
-	{
-		supportLink,
-	}
-);
-
 /**
  * @param {Error?} [error] The error.
+ * @param {string} supportLink The support link.
  * @returns {JSX.Element} The element.
  */
-const ErrorAlert = ( { error } ) => {
+const ErrorAlert = ( { error, supportLink } ) => {
 	if ( ! error ) {
 		return null;
 	}
+
+	// Added dummy space as content to prevent children prop warnings in the console.
+	const link = <Link variant="error" href={ supportLink }> </Link>;
+
 	return (
 		<Alert variant="error">
 			{ error?.name === "TimeoutError"
-				? TimeoutErrorMessage
-				: OtherErrorMessage
+				? createLinkMessage(
+					/* translators: %1$s and %2$s expand to an opening/closing tag for a link to the support page. */
+					__( "A timeout occurred, possibly due to a large number of posts or terms. In case you need further help, please take a look at our %1$sSupport page%2$s.", "wordpress-seo" ),
+					link
+				)
+				: createLinkMessage(
+					/* translators: %1$s and %2$s expand to an opening/closing tag for a link to the support page. */
+					__( "Something went wrong. In case you need further help, please take a look at our %1$sSupport page%2$s.", "wordpress-seo" ),
+					link
+				)
 			}
 		</Alert>
 	);
 };
 
+/**
+ * @param {ContentType?} [contentType] The selected content type.
+ * @param {Term?} [term] The selected term.
+ * @returns {{contentType: string, taxonomy?: string, term?: string}} The score query parameters.
+ */
+const getScoreQueryParams = ( contentType, term ) => { // eslint-disable-line complexity
+	const params = {
+		contentType: contentType?.name,
+	};
+	if ( contentType?.taxonomy?.name && term?.name ) {
+		params.taxonomy = contentType.taxonomy.name;
+		params.term = term.name;
+	}
+
+	return params;
+};
+
+/**
+ * @param {?{scores: Score[]}} [data] The data.
+ * @returns {?Score[]} scores The scores.
+ */
+const prepareScoreData = ( data ) => data?.scores;
```

Comment:
These might be moved to the dataProvider or props?

Reply:
Maybe the data formatter?